internal/controller/factory/statefulset_test.go

```diff
@@ -180,11 +178,6 @@ var _ = Describe("CreateOrUpdateStatefulSet handler", func() {
 
 			By("Checking the extraArgs", func() {
 				Expect(statefulSet.Spec.Template.Spec.Containers[0].Args).To(Equal(generateEtcdArgs(&etcdcluster)))
-				By("Checking args are sorted", func() {
```

Comment:
They are not all sorted anymore, so remove the test

Reply:
I think tests should check that extraArgs in `statefulSet.Spec.Template.Spec.Containers[0].Args` are sorted. So that pods won't be recreated on the next reconciliation

Reply:
Currently, it doesn't make much sense since `spec.Options` field is empty, so we will sort a slice with only one element.

Reply:
Updated the test, but I don't really like the approach :)